OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -163,8 +163,8 @@ public virtual CellContents UpdateDirtyTile(CPos c)
 			NetWorth += t.Density * type.Info.ValuePerUnit;
 
 			var sprites = type.Variants[t.Variant];
-			var frame = int2.Lerp(0, sprites.Length - 1, t.Density - 1, type.Info.MaxDensity);
-			t.Sprite = sprites[frame];
+			var frame = int2.Lerp(0, sprites.Length, t.Density, type.Info.MaxDensity);
+			t.Sprite = sprites[frame - 1];
```

Comment:
Math-wise, the correct code should be:
```csharp
var frame = int2.Lerp(0, sprites.Length - 1, t.Density, type.Info.MaxDensity);
t.Sprite = sprites[frame];
```
If this doesn't work, then I expect the problem is actually in the values that `t.Density` are set to.